File: solutions/buy_sell_stock_multi/buy_sell_stock_multi.cpp
#include <iostream>
#include <vector>
using namespace std;

int maxProfit(vector<int>& prices) {
    if (prices.size() <= 1) 
        return 0;

    int profit = 0;
    for (int i = 1; i < prices.size(); i++) {
        int diff = prices[i] - prices[i - 1];
        if (diff > 0)
            profit += diff;
    }
    return profit;
}

int main() {
    vector<int> prices = {7, 1, 5, 3, 6, 4};
    cout << "Max Profit is: " << maxProfit(prices) << endl; // Expected: 7
    return 0;
}

File: solutions/buy_sell_stock_multi/buy_sell_stock_multi.py
def maxProfit(prices):
    if len(prices) < 2:
        return 0

    profit = 0
    for i in range(1, len(prices)):
        diff = prices[i] - prices[i-1]
        if diff > 0:
            profit += diff
    return profit

# Test
print(maxProfit([7,1,5,3,6,4]))  # Expected output: 7
